packages/rs-drive-abci/src/execution/engine/finalize_block_proposal/v0/mod.rs

```diff
@@ -223,7 +223,11 @@ where
 
         let mut drive_cache = self.drive.cache.write().unwrap();
 
+        // Update global cache with updated contracts
+        drive_cache.cached_contracts.merge_block_cache();
+        // This is unnecessary since we clear block cache before every proposal execution
         drive_cache.cached_contracts.clear_block_cache();
+        drop(drive_cache);
```

Comment:
No need to drop here.

Reply:
Yeah, in this specific case we at the end of function and this is unnecessary, but it's good practice to drop write locks are you done with it since you may move you code later.

Reply:
Sam don't care much so we keep it